service/storage_proxy.cc

```diff
@@ -2447,8 +2447,12 @@ void storage_proxy::maybe_update_view_backlog_of(gms::inet_address replica, std:
     }
 }
 
-db::view::update_backlog storage_proxy::get_view_update_backlog() const {
-    return _max_view_update_backlog.add_fetch(this_shard_id(), get_db().local().get_view_update_backlog());
+db::view::update_backlog storage_proxy::get_view_update_backlog_for_response() const {
+    return _max_view_update_backlog.add_fetch_for_response(this_shard_id(), get_db().local().get_view_update_backlog());
+}
+
+std::optional<db::view::update_backlog> storage_proxy::get_view_update_backlog_for_gossip(std::optional<db::view::update_backlog> last_backlog) const {
+    return _max_view_update_backlog.add_fetch_for_gossip(this_shard_id(), get_db().local().get_view_update_backlog(), std::move(last_backlog));
 }
```

Comment:
Gossiper runs on shard 0, so the shard parameter is unneeded. It would make more sense to check the shard and do on_internal_error is it is not 0.

Reply:
Added